service/src/main/java/io/camunda/service/AuthorizationServices.java

```diff
@@ -46,6 +50,29 @@ public SearchQueryResult<AuthorizationEntity> search(final AuthorizationQuery qu
     return authorizationSearchClient.searchAuthorizations(query, authentication);
   }
 
+  public Set<String> fetchAssignedPermissions(
+      final String ownerType,
+      final String ownerId,
+      final AuthorizationResourceType resourceType,
+      final String resourceId) {
+    final SearchQueryResult<AuthorizationEntity> result =
+        search(
+            SearchQueryBuilders.authorizationSearchQuery(
+                fn ->
+                    fn.filter(
+                            f ->
+                                f.resourceType(resourceType.name())
+                                    .resourceKey(
+                                        StringUtils.isNoneEmpty(resourceId) ? resourceId : null)
```

Comment:
❓ instead of adding a dependency here would something like this also have the same outcome that you're after (checking for empty)
```suggestion
                                        resourceId.isEmpty() ? null : resourceId)
```